src/librustc_typeck/check/_match.rs

```diff
@@ -476,11 +484,12 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
             };
         }
 
+        // We won't diverge unless the discriminant or all arms diverge.
+        self.diverges.set(discrim_diverges | all_pats_diverge | all_bodies_diverge);
```

Comment:
Not that I care *too much* here, but I guess plausibly you could have 50% of the patterns diverging and the other 50% of the bodies diverging, right? 

i.e., the proper test is `discrim_diverges | all_arms_diverge`, where `all_arms_diverge = pat_diverges | body_diverges`